openedx/core/djangoapps/profile_images/tests/test_views.py

```diff
@@ -0,0 +1,228 @@
+
+from student.tests.factories import UserFactory
+
+from ...user_api.accounts.api import set_has_profile_image, get_profile_image_names
+from ...user_api.accounts.helpers import get_profile_image_storage
+from ..images import generate_profile_images, ImageValidationError
+from .helpers import make_image_file
+
+TEST_PASSWORD = "test"
+
+
+class ProfileImageEndpointTestCase(APITestCase):
+    """
+    Base class / shared infrastructure for tests of profile_image "upload" and
+    "remove" endpoints.
+    """
+
+    # subclasses should override this with the name of the view under test, as
+    # per the urls.py configuration.
+    _view_name = None
+
+    def setUp(self):
+        super(ProfileImageEndpointTestCase, self).setUp()
+        self.user = UserFactory.create(password=TEST_PASSWORD)
+        self.url = reverse(self._view_name, kwargs={'username': self.user.username})
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.storage = get_profile_image_storage()
+        # this assertion is made here as a sanity check because all tests
+        # assume user.profile.has_profile_image is False by default
+        self.assertFalse(self.user.profile.has_profile_image)
+
+    def tearDown(self):
+        for name in get_profile_image_names(self.user.username).values():
+            self.storage.delete(name)
+
+    def check_images(self, exist=True):
+        """
+        If exist is True, make sure the images physically exist in storage
+        with correct sizes and formats.
+
+        If exist is False, make sure none of the images exist.
+        """
+        for size, name in get_profile_image_names(self.user.username).items():
+            if exist:
+                self.assertTrue(self.storage.exists(name))
+                img = Image.open(self.storage.path(name))
+                self.assertEqual(img.size, (size, size))
+                self.assertEqual(img.format, 'JPEG')
+            else:
+                self.assertFalse(self.storage.exists(name))
+
+    def check_response(self, response, expected_code, expected_message=None):
+        """
+        Make sure the response has the expected code, and if that isn't 200,
+        optionally check the correctness of a developer-facing message.
+        """
+        self.assertEqual(expected_code, response.status_code)
+        if expected_code == 204:
+            self.assertIsNone(response.data)
+        elif expected_message is not None:
+            self.assertEqual(response.data.get('developer_message'), expected_message)
+
+    def check_has_profile_image(self, has_profile_image=True):
+        """
+        Make sure the value of self.user.profile.has_profile_image is what we
+        expect.
+        """
+        # it's necessary to reload this model from the database since save()
+        # would have been called on another instance.
+        profile = self.user.profile.__class__.objects.get(user=self.user)
```

Comment:
the indirection avoids importing `UserProfile` explicitly, which would create a dependency on common.